### DISCUSSION TITLE
PDBWriter issue

### DISCUSSION THREAD

**Discussion by icamps** [MDAnalysis/mdanalysis#4376] [Q&A]
Hi,

I am using the following code to export my trajectory as a multi-model PDB file:

```
complexx_temp = u_aligned.select_atoms(sel_prot, sel_lig)
with MDAnalysis.Writer("out_" + fname_root + "_PDBtrajectories.pdb", bonds=None, multiframe=True) as W:
    for ts in u_aligned.trajectory:
        W.write(complexx_temp)
```

Then, I use [PLIP](https://plip-tool.biotec.tu-dresden.de/plip-web/plip/index) to calculate the interactions between the ligand and the protein.

The problem here is that the PLIP program is complaining about the PDB final format.

My DCD/PSF files have the complex system information as segment IDs (PROA/HETA) chains (A for protein, B for ligand), residue names (protein ok, and LIG for ligand).

The PDB output from MDAnalysis (v2.6) is in the form:

```
TITLE     MDANALYSIS FRAMES FROM 0, STEP 1: Created by PDBWriter
CRYST1  144.632  144.632  144.632  90.00  90.00  90.00 P 1           1
MODEL        1
ATOM      1  N   MET X   1     -28.647 -45.944  26.553  1.00  0.00      PROA    
ATOM      2  HT1 MET X   1     -29.677 -46.064  26.469  1.00  0.00      PROA    
... 
ATOM   8953  H20 LIG X 999     -57.268 -41.180   4.867  1.00  0.00      HETA    
ATOM   8954  LP1 LIG X 999     -65.187 -35.766   2.503  1.00  0.00      HETA  
```  

Is it possible to check/fix this PDB output by adding the information already present in the DCD/PSF files?

Thanks in advance.

Camps

**Comment by RMeli** [comment]
Hello. You can manually control/modify the different attributes of your `complexx_temp` atom group by setting the corresponding [topology attributes](https://userguide.mdanalysis.org/stable/topology_system.html#topology-attributes). If I understand correctly, the problem is the `PROA` and `HETA` tags that show up in the elements column of the PDB file? Where do they appear in your atom group?

As a first workaround, I'd try to [add an `elements` attribute](https://userguide.mdanalysis.org/stable/examples/constructing_universe.html#Adding-topology-attributes) consisting of empty strings to see if that can circumvent the issue.

---

FYI, there is a tool for protein-ligand interactions, [ProLIF](https://prolif.readthedocs.io/en/stable/), that builds on top of MDAnalysis. I haven't used it and I don't know how it compares with PLIP in practice (in theory, there is a [comparison table](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00548-6/tables/1)), but should circumvent the PDB output all together.

**Reply by icamps** [reply]
Hi Rocco.

Definitively, this is an issue with the atom type field.

Unfortunately, I am not seen how to change/add that info to my universe.

About ProLIF I will take a look on it.

**Reply by RMeli** [reply]
@icamps, I was suggesting to try the following, as a workaround:
```python
# Define n_atoms
complexx_temp.add_TopologyAttr('element', [''] * n_atoms)
```

If you could provide a minimal working example of your issue we could have a deeper look.

**Reply by icamps** [reply]
After adding the piece of code you recommended, I got the error:

`AttributeError: AtomGroup has no attribute add_TopologyAttr.`

Here are the links to the [PSF](https://drive.google.com/file/d/18L0h-TsZCTOKx4Rs_oxJ_Tqhv5ULpSVr/view?usp=drive_link) and [DCD](https://drive.google.com/file/d/1zCaJv4O-yR_Em_ypl-d15jZZy0hLQl6t/view?usp=sharing) files I am using (in case you can take a look).

**Reply by RMeli** [reply]
My bad (sorry, I went by memory...)! I think it should be
```python
u_aligned.add_TopologyAttr('element', [''] * u.atoms.n_atoms)
```
instead.

Thanks for providing the files, I'll have a look.

**Reply by icamps** [reply]
Hi @RMeli, I just create a new python environment using PYENV (instead CONDA), and now everything is working fine.

Sorry for taking your time and @orbeckst time.

**Reply by orbeckst** [reply]
Did @RMeli ’s suggestion work for you or did the problem just go away without any changes? It is very useful for anyone else reading this discussion later to know exactly what the solution is. 

Btw I doubt that it makes a difference how you install. Check that you have the same version of MDA and you really used exactly the same code.

**Reply by icamps** [reply]
I didn't use the last @RMeli suggestion. Just removed CONDA and moved to PYENV. The original script ran fine.

I really don't understand also, but my CONDA get messy very easier.

**Reply by RMeli** [reply]
> but my CONDA get messy very easier.

@icamps if you are not doing so already, I'd strongly suggest to use separate environments for different projects, and re-create new environment when you want to update a large chunk of the dependencies.

**Comment by orbeckst** [comment]
What should the output PDB look like, instead of the example that you gave?
